src/main/java/co/elastic/logstash/filters/elasticintegration/geoip/GeoIpDatabaseAdapter.java

```diff
@@ -52,6 +54,16 @@ public AsnResponse getAsn(InetAddress inetAddress) {
         return getResponse(inetAddress, this.databaseReader::tryAsn);
     }
 
+    /* @Override // neither available nor reachable until Elasticsearch 8.14 */
```

Comment:
review note: _yes_ this annotation is commented out; it should be un-commented once we are exclusively building against Elasticsearch 8.14+.

Unfortunately, since these methods do not exist when compiling against Elasticsearch < 8.14, an `@Override` annotation would prevent compilation.

Since it is commented out, we do not have a compile-time guarantee that we are _overriding_ the method from the interface (as opposed to overloading it).

I could not think of a way to dynamically implement the upstream `GeoIpDatabase` without resorting to very complex reflection-based proxy dispatch.